pkg/docker/docker.go

```diff
@@ -126,6 +126,62 @@ type RunContainerOptions struct {
 	CapDrop          []string
 }
 
+// asDockerConfig converts a RunContainerOptions into a Config understood by the
+// docker client library.
+func (rco RunContainerOptions) asDockerConfig() docker.Config {
+	return docker.Config{
+		Image: getImageName(rco.Image),
+		User:  rco.User,
+		// TODO why do we call runContainerTar with a "partial"
+		// docker.Config?
+
+		Env: rco.Env,
+
+		// TODO review how callers use these options to pass stdin.
+		// Probably they are not necessary, as they're related to
+		// attaching to the container (that is often unnecessary).
```

Comment:
yeah so basically we need to attach stdin for `assemble` (to stream in the content).  

we need to attach stdout/stderr for `run` and for when we run the resulting image after creating it (so we can see the logs)

presumably we'd want to attach stderr when running `save-artifacts` in case the save-artifacts script uses it.

I don't know what other conditions we might need to worry about (ie the on-build flows and layered build flows)